flash/transport.go

```diff
@@ -0,0 +1,53 @@
+package flash
+
+type Command byte
+
+const (
+	CmdRead            Command = 0x03 // Single Read
+	CmdQuadRead                = 0x6B // 1 line address, 4 line data
+	CmdReadJedecID             = 0x9f
+	CmdPageProgram             = 0x02
+	CmdQuadPageProgram         = 0x32 // 1 line address, 4 line data
+	CmdReadStatus              = 0x05
+	CmdReadStatus2             = 0x35
+	CmdWriteStatus             = 0x01
+	CmdWriteStatus2            = 0x31
+	CmdEnableReset             = 0x66
+	CmdReset                   = 0x99
+	CmdWriteEnable             = 0x06
+	CmdWriteDisable            = 0x04
+	CmdEraseSector             = 0x20
+	CmdEraseBlock              = 0xD8
+	CmdEraseChip               = 0xC7
+)
+
+type Error uint8
+
+const (
+	_                          = iota
+	ErrInvalidClockSpeed Error = iota
+	ErrInvalidAddrRange
+)
+
+func (err Error) Error() string {
+	switch err {
+	case ErrInvalidClockSpeed:
+		return "invalid clock speed"
+	case ErrInvalidAddrRange:
+		return "invalid address range"
+	default:
+		return "unspecified error"
```

Comment:
Nit: it would be helpful to include the source of the error, such as `flash: invalid clock speed`. That should help in the case the error is printed as-is (using `println` for example).

Reply:
good point, added this.